src/stamps/hardcoded.ts

```diff
@@ -1,12 +1,9 @@
-import { StampsConfig } from '../config'
+import { StampsConfigHardcoded } from '../config'
 import { BaseStampManager, StampsManager } from './base'
 
 export class HardcodedStampsManager extends BaseStampManager implements StampsManager {
-  /**
-   * Start the manager in either hardcoded or autobuy mode
-   */
-  async start(config: StampsConfig): Promise<void> {
-    // Hardcoded stamp mode
-    if (config.mode === 'hardcoded') this.stamp = config.stamp
+  constructor(config: StampsConfigHardcoded) {
+    super()
+    this.start(config)
```

Comment:
I believe that Aron suggested you move the `this.start()` call "externally", which I agree with him. To your question what does that mean, it means that the constructor should not start the StampManger itself. The semantic of a constructor is to create a new instance and not to do much more like starting the work of StampManager. Starting should be left to the "instantiator" (for example he might want to start the work of StampManger after some dependencies are met), which in this case is the `index.ts` file, so please move the `this.start()` call there. Also IMHO the `this.start()` does not really need to take in the config object, instead I would pass it to the constructor (which you already do 👍), but then store it into some private property that then the instance uses in its lifecycle.